src/MagnumPlugins/DrFlacAudioImporter/DrFlacImporter.cpp

```diff
@@ -74,12 +103,30 @@ void DrFlacImporter::doOpenData(Containers::ArrayView<const char> data) {
         return;
     }
 
-    Containers::Array<char> tempData(samples*sizeof(int32_t));
+    _frequency = handle->sampleRate;
+    _format = flacFormatTable[numChannels][normalizedBitsPerSample];
 
-    drflac_read_s32(handle, samples, reinterpret_cast<int32_t*>(tempData.begin()));
+    Containers::Array<char> tempData(samples*sizeof(Int));
+    drflac_read_s32(handle, samples, reinterpret_cast<Int*>(tempData.begin()));
     drflac_close(handle);
 
-    _data = std::move(tempData);
+    switch(normalizedBitsPerSample)
+    {
+        case 1: {
+            _data = convert32PCM(tempData, samples, sizeof(UnsignedByte));
+
+            // Convert to unsigned
+            for(char& item : _data) {
+                item = item - 128;
+            }
+            break;
+        }
+        case 2: {
+            _data = convert32PCM(tempData, samples, sizeof(Short));
+            break;
+        }
+    }
```

Comment:
I'd prefer just a simple if/else as it is way shorter, then also asserting that there's no other value ever taken:

```cpp
if(normalizedBytesPerSample == 1) {
    _data = convert32PCM(tempData, samples, sizeof(UnsignedByte);
    for(char& item: _data) item -= 128;
} else if(normalizedBytesPerSample == 2) {
    _data = convert32PCM(tempData, samples, sizeof(Short));
} else CORRADE_ASSERT_UNREACHABLE();
```

Isn't the `convert32PCM()` equivalent to `_data = std::move(tempData);` in the first case anyway? Or am I missing something?